### PR TITLE
[FIX] dnspod with modify, delete and list dns record from dnspod.cn

### PR DESCRIPTION
**Pull request by noahzaozao**
https://github.com/OCA/infrastructure-dns/pull/4

### CODE REVIEW

File: connector_dns_dnspod/__init__.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (elicoidal) [inline connector_dns_dnspod/__init__.py]: @noahzaozao   Thanks for your contribution to the OCA. 
Let me introduce to you some basic information about contribution in OCA and I will review other points in more comments inside the code..
In general, please follow the principles: 
- Contribution guidelines: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
- Please follow the OCA README template:
  https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
- We recommend to include a screenshot as well to help new comers (nice to have)
- Please use simple headers in all py files following OCA templates
  https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3
- Please sort all imports in alphabetical order following: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

in this particular example, please use short header in all py files

File: connector_dns_dnspod/__init__.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+from . import backend
+from . import dnspod
```
Comment (elicoidal) [inline connector_dns_dnspod/__init__.py:6]: you should probably protect teh import

File: connector_dns_dnspod/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{'name': 'DNSpod connector',
+ 'version': '0.1',
```
Comment (elicoidal) [inline connector_dns_dnspod/__openerp__.py]: 8.0.1.0.0

File: connector_dns_dnspod/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{'name': 'DNSpod connector',
```
Comment (elicoidal) [inline connector_dns_dnspod/__openerp__.py]: DNSPod connector (China)

File: connector_dns_dnspod/__init__.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline connector_dns_dnspod/__init__.py:1]: You should create a README file

File: connector_dns_dnspod/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ 'version': '0.1',
+ 'category': 'Connector',
+ 'depends': ['connector_dns'],
+ 'author': 'Elico Corp,Odoo Community Association (OCA)',
+ 'license': 'AGPL-3',
+ 'website': 'https://www.elico-corp.com',
+ 'images': [],
+ 'demo': [],
```
Comment (elicoidal) [inline connector_dns_dnspod/__openerp__.py]: remove this key

File: connector_dns_dnspod/readme.rst [hunk 7]
```diff
@@ -0,0 +1,65 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (elicoidal) [inline connector_dns_dnspod/readme.rst]: capitals for the name

File: connector_dns_dnspod/readme.rst [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    :alt: License
+
+Connector Dns Dnspod
+=====================
```
Comment (elicoidal) [inline connector_dns_dnspod/readme.rst]: line only until the end of the title

File: connector_dns_dnspod/readme.rst [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+To configure this module, you need to:
+
+ * No specific configuration needed.
+
+Usage
+=====
+
+To use this module, you need to:
+1.Create a backend which links to your dnspod.cn.
```
Comment (elicoidal) [inline connector_dns_dnspod/readme.rst]: use "#. " better

File: connector_dns_dnspod/readme.rst [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+2.When you create a domain belongs to the backend,if the domain 
+  export to the dnspod.cn successfully,the state will change to 
+  done,else exception.
+3.Record can be created only in the domain which state is done. 
+
+
+For further information, please visit:
```
Comment (elicoidal) [inline connector_dns_dnspod/readme.rst]: remove this part

File: connector_dns_dnspod/readme.rst [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+ * https://www.odoo.com/forum/help-1
+
+Known issues / Roadmap
+======================
+
```
Comment (elicoidal) [inline connector_dns_dnspod/readme.rst]: This connector is only compatible with dnspod.cn and not dnspod.com which has a different API.

File: connector_dns_dnspod/readme.rst [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+Credits
+=======
+
+
+Contributors
+------------
+
+* Liu Lixia <liu.lixia@elico-corp.com>
+* Augustin Cisterne-Kaas
+
```
Comment (elicoidal) [inline connector_dns_dnspod/readme.rst]: add yourself and myself

File: connector_dns_dnspod/readme.rst [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+Maintainer
+----------
+
+.. image:: http://odoo-community.org/logo.png
```
Comment (elicoidal) [inline connector_dns_dnspod/readme.rst]: https (x3)

File: connector_dns_dnspod/security/ir.model.access.csv [hunk 14]
```diff
@@ -0,0 +1,10 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+
```
Comment (elicoidal) [inline connector_dns_dnspod/security/ir.model.access.csv]: why the blank lines?

File: connector_dns_dnspod/__init__.py (hunk 5 above)
Comment (noahzaozao) [inline connector_dns_dnspod/__init__.py:1]: you means the README.rst ?

File: connector_dns_dnspod/readme.rst (hunk 7 above)
Comment (noahzaozao) [inline connector_dns_dnspod/readme.rst]: Done

File: connector_dns_dnspod/dnspod.py [hunk 15]
```diff
@@ -0,0 +1,236 @@
...
+
+class DNSPodRecord(models.Model):
+    _inherit = 'dns.record'
+
+    record_id = fields.Integer(string="Record id on dnspod.cn")
+
+    def _type_select_version(self):
+        res = [('A', 'A'), ('CNAME', 'CNAME'), ('MX', 'MX'),
+               ('TXT', 'TXT'), ('NS', 'NS'), ('AAAA', 'AAAA'),
+               ('SRV', 'SRV'), ('Visibile URL', '显性URL'),
```
Comment (elicoidal) [inline connector_dns_dnspod/dnspod.py:178]: Visible URL

**Comment by noahzaozao** [comment]
@elicoidal I need some help with fix the errors in travis , thanks.

**Comment by noahzaozao** [comment]
@elicoidal I fixed the travis error for this module , and now it's waiting for the https://github.com/OCA/infrastructure-dns/pull/4 merge

**Comment by elicoidal** [comment]
@noahzaozao I think the base module should be merged now. We should move forward with this one.
There are some comments to be attended. Not sure whether this is the latest version of the module though.

**Comment by noahzaozao** [comment]
@elicoidal as I analyzed weeks ago , the API change between old v8.0 and now v8.0 are huge , I have explained to Seb , so it takes some time to migration to latest connector_dns and there may some issues of the new connector_dns need to be fix.

**Comment by elicoidal** [comment]
Do you prefer to close this PR or maintain it for modification later?

File: oca_dependencies.txt [hunk 16]
```diff
@@ -1,2 +1,3 @@
+connector
```
Comment (seb-elico) [inline oca_dependencies.txt:1]: Already a dependency of `infrastructure-dns`

**Pull request closed without merging**